src/Exceptions/ApiLimitReachedException.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace ParcelTrap\Exceptions;
+
+use ParcelTrap\Contracts\Driver;
+use Throwable;
+
+class ApiLimitReachedException extends ParcelTrapDriverException
+{
+    public function __construct(
+        private readonly int $limit,
+        private readonly string $period,
```

Comment:
If these are readonly, may as well make them public and get rid of the getters 👍🏻
```suggestion
        public readonly int $limit,
        public readonly string $period,
```